util/canonicaljson.py

```diff
@@ -15,5 +15,5 @@ def canonicalize(json_obj):
         sorted_obj = sorted({key: canonicalize(val) for key, val in json_obj.items()}.items())
         return collections.OrderedDict(sorted_obj)
     elif isinstance(json_obj, (list, tuple)):
-        return [canonicalize(val) for val in json_obj]
+        return sorted([canonicalize(val) for val in json_obj])
```

Comment:
This is incorrect: It would alter the order of items in the list, and result in producing different JSON data

Reply:
Ah, this is a good point. It still fixes the bug, though! Maybe I could add a helper function in this module that recursively sorts lists if you don't care about the order? There really isn't an equivalent for sets in JSON.

Reply:
Sure. I'd have it take in a sort key as well, so you can at least define the order for sorting. Then, make sure the sorting is off for any registry usage of this method

Reply:
While I expect this to rarely be used, I added it as a kwarg.